frontend/pages/Homepage/cards/Software/Software.tsx

```diff
@@ -221,29 +235,26 @@ const Software = ({
     }
   };
 
-  const onModalSoftwareQueryChange = debounce(
+  const onModalSoftwareQueryChange = useDebouncedCallback(
     async ({ pageIndex, searchQuery }: ITableQueryProps) => {
       setModalSoftwareSearchText(searchQuery);
 
       if (pageIndex !== modalSoftwarePageIndex) {
         setModalSoftwarePageIndex(pageIndex);
+      } else {
+        try {
+          // TODO: count loading UX?
```

Comment:
I know the answer to this one. We should treat the count loading state like we treat the "Refetch" host details loading state. Same spinning circle icon with the text "Loading software count" or "Loading count" based on space available.